File: selection_sort/selectionSort.go
package main

import "fmt"

func findMinimal(array []int) int {
	var pos int

	for i, n := range array[1:] {
		if array[pos] > n {
			pos = i + 1
		}
	}
	return pos
}

func selectionSort(array *[]int) {
	var i, j int

	for ; i<len((*array)); i++ {
		j = findMinimal((*array)[i:]) + i
		(*array)[i], (*array)[j] = (*array)[j], (*array)[i]
	}
}

func main() {
	var test = []int{8, 5, 2, 9, 5, 6, 3}
	var example = []int{2, 3, 5, 5, 6, 8, 9}

	selectionSort(&test)
	fmt.Println(test, "vs", example)
}

File: selection_sort/selection_sort.py
def find_smaller(array):
    minimal = 0
    for i in range(1, len(array)):
        if array[minimal] > array[i]:
            minimal = i
    return minimal

def swap(i, j, array):
    array[i], array[j] = array[j], array[i]

def selectionSort(array):
    for i in range(len(array)):
        j = find_smaller(array[i:]) + i
        swap(i, j, array)
    return array

if __name__ == "__main__":
    test = [8, 5, 2, 9, 5, 6, 3]
    example = [2, 3, 5, 5, 6, 8, 9]
    print(selectionSort(test), "vs", example)
